MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift

```diff
@@ -364,4 +377,505 @@ final class MSALNativeAuthPublicClientApplicationTest: XCTestCase {
+        cacheAccessorMock.expectedMSIDTokenResult = tokenResult
+        
+        delegateVerifyCode.expectedUserAccountResult = userAccountResult
+        let signInController = MSALNativeAuthSignInController(clientId: clientId,
+                                                                         signInRequestProvider: signInRequestProviderMock,
+                                                                         tokenRequestProvider: tokenRequestProviderMock,
+                                                                         cacheAccessor: cacheAccessorMock,
+                                                                         factory: authResultFactoryMock,
+                                                                         signInResponseValidator: signInResponseValidatorMock,
+                                                                         tokenResponseValidator: tokenResponseValidatorMock)
+        let controllerFactory = MSALNativeAuthControllerProtocolFactoryMock(signInController: signInController)
+        
+        sut = MSALNativeAuthPublicClientApplication(
+            controllerFactory: controllerFactory,
+            inputValidator: MSALNativeAuthInputValidator(),
+            internalChallengeTypes: []
+        )
+        
+        // Correlation Id is validated internally against contextMock on both initiate and challenge
+        sut.signInUsingPassword(username: "username", password: "password", scopes: ["scope1", "scope2"], correlationId: correlationId, delegate: delegatePasswordStart)
+        
+        wait(for: [expectationPasswordStart])
+        
+        // Correlation Id is validated internally against expectedTokenParams
+        delegatePasswordStart.newSignInCodeRequiredState?.submitCode(code: "1234", delegate: delegateVerifyCode)
+        wait(for: [expectationVerifyCode])
+        
+        // User account result is validated internally against expectedUserAccountResult
+        XCTAssertTrue(delegateVerifyCode.onSignInCompletedCalled)
+    }
+    
+    // SignIn Code
+    // Testing SignInInitiate -> SignInChallenge -> SignInToken with Code
+    
+    func testSignInCode_correlationId_whenSetOnStart_itCascadesToAll() {
+        let expectationCodeStart = expectation(description: "Sign In Code Start")
+        let delegateCodeStart = SignInCodeStartDelegateSpy(expectation: expectationCodeStart)
+        let expectationVerifyCode = expectation(description: "Sign In Verify Code")
+        let delegateVerifyCode = SignInVerifyCodeDelegateSpy(expectation: expectationVerifyCode)
+        
+        let signInRequestProviderMock = MSALNativeAuthSignInRequestProviderMock()
+        signInRequestProviderMock.mockInitiateRequestFunc(MSALNativeAuthHTTPRequestMock.prepareMockRequest())
+        signInRequestProviderMock.mockChallengeRequestFunc(MSALNativeAuthHTTPRequestMock.prepareMockRequest())
+        signInRequestProviderMock.expectedUsername = "username"
+        
```

Comment:
Same as the other comment, the context validation is missing in this test

Reply:
Indeed, fixed